appveyor.yml

```diff
@@ -30,11 +30,16 @@ install:
     - appveyor-retry stack exec -- pacman -S autoconf automake-wrapper make patch python tar --noconfirm
 
 build_script:
+    # Boot and configure ghc source tree
+    - cd ..
+    - python boot && configure --enable-tarballs-autodownload
```

Comment:
Why explicit booting and configuration? Doesn't `-c` flag work?

Reply:
For some reason, with my changes `-c` does not seem to run the boot/configure scripts anymore, or not in time for reading some settings apparently. I could not figure out why the boot/configure rules don't fire.

Reply:
Let's try to figure this out. The dependency mechanism is via the `configFile`. It is built by `configureRules` and is `need`ed via `lookupValueOrError configFile` in `Oracles.Setting`.

> not in time for reading some settings apparently

Which setting is it?

Reply:
Sorry for the delay, takes a time to go through all the feedback. I went back to the travis CI run where we boot with 8.2, last commit before I removed `-c` and did the configure phase explicitly, you can see the error [right here](https://travis-ci.org/snowleopard/hadrian/jobs/356346912#L1564). So this branch somehow requires data from the cabal files very early on (apparently before we get a chance to fire the configure rules), but those cabal files are supposed to be generated from `.cabal.in` files during configure.

Reply:
Ah, I think I know what's going on! We haven't included `compiler/ghc.cabal` into the result of `configureRules`. We need to modify this line:

https://github.com/snowleopard/hadrian/blob/master/src/Rules/Configure.hs#L15

If you add `compiler/ghc.cabal` into the list of resulting files I bet it will work. Note that this will also add a dependency on `compiler/ghc.cabal.in`, so if this file is edited, the configure script will be automatically rerun to produce an up-to-date `compiler/ghc.cabal`.